boxtree/fmm.py

```diff
@@ -43,6 +43,33 @@
 
 # {{{ expansion wrangler interface
 
+# Design considerations:
+#
+# - Making the wrangler contain/permit it to depend the tree (which was previously the case)
+#   forces code caches (say, for translations) to be thrown away every time the FMM is run on
```

Comment:
What do you mean by code caches here?

Reply:
Like sumpy's `wrangler.code.p2m`. Things that ultimately hold the reference to the finished loopy kernel (e.g.) doing the translation.

Reply:
Got it. And `rotation_classes` would be cached here as well? (`translation_classes` depend on the tree, so those will have to be cached in taw)